src/main/java/com/force/i18n/grammar/impl/GrammaticalTermMapImpl.java

```diff
@@ -0,0 +1,196 @@
+            for (String name : termsToInclude) {
+                GrammaticalTerm term = map.get(name);
+                if(term != null) {
+                    if (term instanceof Noun) term = dictionary.getNounOverride((Noun)term);
+                    if (!first) {
+                        out.append(',');
+                    } else {
+                        first = false;
+                    }
+                    writeJsonTerm(out, renamingProvider, term, dictionary.getLanguage());
+                    wrote.add(name);
+                }
+            }
+            termsToInclude.removeAll(wrote);
+        } else {
+            writeJson(out, renamingProvider, dictionary.getLanguage());
+        }
+        out.append('}');
+    }
+    
+    private void writeJson(Appendable out, RenamingProvider renamingProvider, HumanLanguage lang) throws IOException {
+        boolean first = true;
+        for (GrammaticalTerm term : map.values()) {
+            if (!first) {
+                out.append(',');
+            } else {
+                first = false;
+            }
+            writeJsonTerm(out, renamingProvider, term, lang);
+        }
+    }
+    
+    private void writeJsonTerm(Appendable out, RenamingProvider renamingProvider, GrammaticalTerm term, HumanLanguage lang) throws IOException {
+        if (renamingProvider != null && term instanceof Noun && renamingProvider.useRenamedNouns()) {
+            Noun renamedNoun = renamingProvider.getRenamedNoun(lang, ((Noun)term).getName());
+            if (renamedNoun != null) term = renamedNoun;
+        }
+        out.append('\"').append(term.getName().toLowerCase()).append("\":");
+        term.toJson(out);
+    }
+
+    @Override
+    public Set<String> keySet() {
+        return map.keySet();
+    }
+
+    @Override
+    public T get(String name) {
+        return map.get(name);
+    }
+
+
+    @Override
+    public boolean containsKey(String name) {
+        return map.containsKey(name);
+    }
+
+    @Override
+    public Set<Map.Entry<String, T>> entrySet() {
+        return map.entrySet();
+    }
+
+    @Override
+    public Collection<T> values() {
+        return map.values();
+    }
+
+    @Override
+    public void put(String k, T v) {
+        if(isSkinny)
+            throw new RuntimeException("This map is not able to modify");
+        map.put(k,v);
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Override 
+    public void putAll(GrammaticalTermMap<T> other) {
+        if(isSkinny)
+            throw new RuntimeException("This map is not able to modify");
+        map.putAll(((GrammaticalTermMapImpl<T>)other).map);
+    }
+    
+    @Override
+    public boolean isEmpty() {
+        return map.isEmpty();
+    }
+
+    /**
+     * Override default serializer to avoid any duplicated in the serialized map.
+     * @param in
+     * @throws IOException
+     */
+    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
```

Comment:
might need `@SuppressWarnings("unchecked")` here
```suggestion
    @SuppressWarnings("unchecked")
    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
```